homeassistant/components/template/sensor.py

```diff
@@ -227,7 +232,42 @@ async def async_added_to_hass(self):
     @callback
     def _update_state(self, result):
         super()._update_state(result)
-        self._attr_native_value = None if isinstance(result, TemplateError) else result
+        if isinstance(result, TemplateError):
+            self._attr_native_value = None
+            return
+
+        if result is None or self.device_class not in (
+            SensorDeviceClass.DATE,
+            SensorDeviceClass.TIMESTAMP,
+        ):
+            self._attr_native_value = result
+            return
+
+        if self.device_class == SensorDeviceClass.TIMESTAMP:
```

Comment:
We do exactly the same check in `TriggerSensorEntity`, can we extract this to a helper?
```
def parse_timestamp(timestamp):
    ...

        if self.device_class == SensorDeviceClass.TIMESTAMP:
            self._attr_native_value = parse_timestamp(state)
```

Reply:
Done!